components/provider_ppucd/src/parse_ppucd.rs

```diff
@@ -0,0 +1,1107 @@
+                }
+            }
+        }
+        if !&uniset_builder.is_empty() {
+            let uniset = uniset_builder.build();
+            m.insert(canonical_name, uniset);
+        }
+    }
+
+    m
+}
+
+fn get_enum_prop_unisets<'s>(
+    enum_prop_aliases: &HashMap<&'s str, HashSet<&'s str>>,
+    enum_val_aliases: &HashMap<&'s str, HashMap<&'s str, HashSet<&'s str>>>,
+    code_points: &HashMap<u32, HashMap<&'s str, &'s str>>,
+) -> HashMap<&'s str, UnicodeSet> {
+    let mut m: HashMap<&str, HashMap<&str, UnicodeSetBuilder>> = HashMap::new();
+    let enum_prop_mappings: HashMap<&str, &str> = aliases_as_canonical_mappings(enum_prop_aliases);
+
+    let enum_val_mappings: HashMap<&str, HashMap<&str, &str>> =
+        get_enum_val_canonical_mapping(enum_val_aliases);
+
+    for (canonical_prop_name, all_prop_name_aliases) in enum_prop_aliases {
+        for (code_point, code_point_prop_key_vals) in code_points {
+            let code_point_prop_names: HashSet<&str> =
+                code_point_prop_key_vals.keys().copied().collect();
+            if !all_prop_name_aliases.is_disjoint(&code_point_prop_names) {
+                for prop_name in all_prop_name_aliases.intersection(&code_point_prop_names) {
+                    let val_name: &str = code_point_prop_key_vals.get(prop_name).unwrap();
+                    let canonicalized_val_name: &str = enum_val_mappings
+                        .get(canonical_prop_name)
+                        .unwrap()
+                        .get(val_name)
+                        .unwrap();
+
+                    if !m.contains_key(canonical_prop_name) {
+                        m.insert(canonical_prop_name, HashMap::new());
+                    }
+
+                    if !m
+                        .get(canonical_prop_name)
+                        .unwrap()
+                        .contains_key(canonicalized_val_name)
+                    {
+                        let result_prop_val_builder_map: &mut HashMap<&str, UnicodeSetBuilder> =
+                            m.get_mut(canonical_prop_name).unwrap();
+                        result_prop_val_builder_map
+                            .insert(canonicalized_val_name, UnicodeSetBuilder::new());
+                    }
+
+                    let enum_val_uniset_builder: &mut UnicodeSetBuilder = m
+                        .get_mut(canonical_prop_name)
+                        .unwrap()
+                        .get_mut(canonicalized_val_name)
+                        .unwrap();
+                    enum_val_uniset_builder.add_char(std::char::from_u32(*code_point).unwrap());
+                }
+            }
+        }
+    }
+
+    let mut result: HashMap<&str, UnicodeSet> = HashMap::new();
+
+    for (canonical_prop_name, prop_val_builder_map) in m {
+        for (canonical_val_name, uniset_builder) in prop_val_builder_map {
+            let enum_val_uniset_name: String =
+                format!("{}={}", canonical_prop_name, canonical_val_name);
```

Comment:
3 options provided by Shane:

1. Use a substring from ppucd.txt itself in the `value;<propname>;<propval>'...` line in the header section
2. Use a TinyString
3. Use a Cow (and construct it with `Cow::Owned` b/c the input is a `String` resulting from `format!`)

Tentative approach:
Go with option 3 (use a Cow), and leave TinyString alternative for #454 optimization issue work. Also, make replacing `format!` a task item to consider in #454.